src/main/java/games/strategy/triplea/ui/logic/RouteCalculator.java

```diff
@@ -162,14 +136,34 @@ public Point getLastEndPoint() {
    * @param ycoords an array of yCoordinates
    * @return a List of corresponding Lines on every possible screen
    */
-  public List<Line> getAllNormalizedLines(final double[] xcoords, final double[] ycoords) {
-    final List<Line> centerLines = getNormalizedLines(xcoords, ycoords);
-    final List<Line> result = new ArrayList<>();
-    for (final Line line : centerLines) {
-      final List<Point[]> allPoints = getAllPoints(line.getP1(), line.getP2());
-      for (final Point[] points : allPoints) {
-        result.add(new Line(points[0], points[1]));
-      }
+  public List<Path2D> getAllNormalizedLines(final double[] xcoords, final double[] ycoords) {
```

Comment:
This method (and subsequently `getNormalizedLines()` and `getPossibleTranslations()`) has zero test coverage.  Would it be easy to add a characterization test using the old implementation to ensure the new implementation hasn't changed the result?  It might require four test cases to cover all the branches in `getPossibleTranslations()`.

Reply:
Characterization Tests could be hard, but luckily I wrote the RouteCalculator class myself so I still remember the algorithm I implemented.
It's probably not the best way to achieve it, but it works.
If you're interested I'd be happy to try to explain it you you.